tests/browser/js/captionsEnactorTests.js

```diff
@@ -211,20 +211,86 @@
          * gpii.uioPlus.captions.createPlayers tests
          **************************************************************************************************************/
 
-        jqUnit.test("Test gpii.uioPlus.captions.createPlayers", function () {
+        jqUnit.asyncTest("Test gpii.uioPlus.captions.createPlayers", function () {
             // setup
             var playerStub = sinon.stub(gpii.uioPlus, "player");
             var expected = ["player1", "player2"];
+            var injected = expected.concat(["player3"]);
             playerStub.onCall(0).returns(expected[0]);
             playerStub.onCall(1).returns(expected[1]);
+            playerStub.onCall(2).returns(injected[2]);
 
-            // test
-            var players = gpii.uioPlus.captions.createPlayers(true);
+            // test existing
+            var model = {captionsEnabled: true};
+            var createdPlayers = gpii.uioPlus.captions.createPlayers(model);
             jqUnit.assertTrue("The gpii.uioPlus.player should have been called for each video found", playerStub.calledTwice);
-            jqUnit.assertDeepEq("Players for each video should have been created", expected, players);
+            jqUnit.assertDeepEq("Players for each video should have been created", expected, createdPlayers.players);
 
-            // cleanup
-            playerStub.restore();
+            // test injected
+            var container = $(".injection");
+            var video = $("<iframe width=\"560\" height=\"315\""
+                + "src=\"https://www.youtube.com/embed/SjnXy0Iplvs\""
+                + "frameborder=\"0\" style=\"border: solid 4px #37474F\""
+                + "allow=\"autoplay; encrypted-media\" allowfullscreen> </iframe>");
+
+            video.ready(function () {
+                jqUnit.assertTrue("The gpii.uioPlus.player should have been called for the injected videos", playerStub.calledThrice);
+                jqUnit.assertDeepEq("The injected player should be added to the set of players.", injected, createdPlayers.players);
+
+                // cleanup
+                container.html("");
+                playerStub.restore();
+                createdPlayers.observer.disconnect();
+
+                jqUnit.start();
+            });
+            container.append(video);
+        });
+
+        jqUnit.asyncTest("Test gpii.uioPlus.captions.createPlayers - removal", function () {
```

Comment:
I get a test failure on this one - 
````

Test gpii.uioPlus.captions.createPlayers - removal: Players for each video should have been created

Expected: | [   {     "videoElm": {}   },   {     "videoElm": {}   },   {     "videoElm": {}   },   {     "videoElm": {       "allowTransparency": "true"     }   } ]
-- | --
[   {     "videoElm": {}   },   {     "videoElm": {}   },   {     "videoElm": {}   } ]
[    {      "videoElm":  {}    },    {      "videoElm":  {}    },    {      "videoElm":  {}   },   {     "videoElm": {       "allowTransparency": "true"     }    }  ]
at Object.assertDeepEq (http://localhost:7357/node_modules/infusion/tests/test-core/jqUnit/js/jqUnit.js:180:23)     at Object.<anonymous> (http://localhost:7357/tests/browser/js/captionsEnactorTests.js:273:20)     at Test.run (http://localhost:7357/node_modules/infusion/tests/lib/qunit/js/qunit.js:207:18)     at http://localhost:7357/node_modules/infusion/tests/lib/qunit/js/qunit.js:365:10     at process (http://localhost:7357/node_modules/infusion/tests/lib/qunit/js/qunit.js:1457:24)     at http://localhost:7357/node_modules/infusion/tests/lib/qunit/js/qunit.js:483:5

````

Reply:
Can reproduce in the standard Windows 10 vagrant box

Reply:
Seems like my search for video elements using just `$("iframe")` was too general. I think the npm test run must add an iframe to the file for reporting or something. I've added a class on those elements that i'm looking for to scope things better.